kaggle_environments/core.py

```diff
@@ -76,20 +77,34 @@ def make(environment, configuration={}, steps=[], debug=False):
         Environment: Instance of a specific environment.
     """
     if has(environment, str) and has(environments, dict, path=[environment]):
-        return Environment(**environments[environment], configuration=configuration, steps=steps, debug=debug)
+        return Environment(**environments[environment], configuration=configuration, steps=steps, logs=logs, debug=debug)
     elif callable(environment):
-        return Environment(interpreter=environment, configuration=configuration, steps=steps, debug=debug)
+        return Environment(interpreter=environment, configuration=configuration, steps=steps, logs=logs, debug=debug)
     elif has(environment, path=["interpreter"], is_callable=True):
-        return Environment(**environment, configuration=configuration, steps=steps, debug=debug)
+        return Environment(**environment, configuration=configuration, steps=steps, logs=logs, debug=debug)
     raise InvalidArgument("Unknown Environment Specification")
 
 
+def act_agent(args):
```

Comment:
Multiprocessing needs to pickle all methods it calls and python can only pickle top-level methods so I moved this up here and passed in all needed state rather than pulling state from the outer closure as before.